lib/Doctrine/DBAL/Platforms/DB2iSeriesPlatform.php

```diff
@@ -0,0 +1,312 @@
+        ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getCreateDatabaseSQL($database)
+    {
+        return "CREATE COLLECTION ".$database;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doModifyLimitQuery($query, $limit, $offset = null)
+    {
+        if ($limit === null && $offset === null) {
+            return $query;
+        }
+
+        $limit = (int) $limit;
+        $offset = (int) (($offset)?:0);
+
+        $orderBy = stristr($query, 'ORDER BY');
+
+        $orderByBlocks = preg_split('/\s*ORDER\s+BY/', $orderBy );
+
+        //Reversing arrays beacause external order by is more important
+        $orderByBlocks = array_reverse($orderByBlocks);
+
+        //Splitting ORDER BY
+        $orderByParts = array();
+        foreach ($orderByBlocks as $orderByBlock){
+            $blockArray   = explode(',', $orderByBlock);
+            foreach ($blockArray as $block) {
+                $block = trim($block);
+                if (!empty($block)) {
+                    $orderByParts[] = $block;
+                }
+            }
+        }
+
+        //Clear ORDER BY
+        foreach ($orderByParts as &$orderByPart) {
+
+            $orderByPart = preg_replace('/ORDER\s+BY\s+([^\)]*)(.*)/', '$1', 'ORDER BY '.$orderByPart);
+        }
+
+        $orderByColumns = array();
+
+        //Split ORDER BY into parts
+        foreach ($orderByParts as &$part) {
+
+            if (preg_match('/(([^\s]*)\.)?([^\.\s]*)\s*(ASC|DESC)?/i', trim($part), $matches)) {
+                $orderByColumns[] = array(
+                    'column'    => $matches[3],
+                    'hasTable'  => ( !empty($matches[2])),
+                    'sort'      => isset($matches[4]) ? $matches[4] : null,
+                    'table'     => empty($matches[2]) ? '[^\.\s]*' : $matches[2]
+                );
+            }
+        }
+
+        //Find alias for each colum used in ORDER BY
+        if ( ! empty($orderByColumns)) {
+            foreach ($orderByColumns as $column) {
+
+                $pattern = sprintf('/%s\.%s\s+(?:AS\s+)?([^,\s)]+)/i', $column['table'], $column['column']);
+                $overColumn = preg_match($pattern, $query, $matches)
+                    ? ($column['hasTable'] ? $column['table']  . '.' : '') . $column['column']
+                    : $column['column'];
+
+                if (isset($column['sort'])) {
+                    $overColumn .= ' ' . $column['sort'];
+                }
+
+                $overColumns[] = $overColumn;
+            }
+        }
+
+        //Replace only first occurrence of FROM with $over to prevent changing FROM also in subqueries.
+        if ( ! $orderBy) {
+            $over = '';
+        }
+        else
+        {
+            $over  = 'ORDER BY ' . implode(', ', $overColumns);
+        }
+
+        $sql = 'SELECT DOCTRINE_TBL.* FROM (SELECT ROW_NUMBER() OVER('.$over.') AS DOCTRINE_ROWNUM, DOCTRINE_TBL1.* '.
```

Comment:
the `DOCTRINE_TBL1.*` needs to come before `DOCTRINE_ROWNUM`, as Doctrine Migrations casts the result of a query using this to a string, the additional column causes the "current migration" to be `1` instead of the correct version.